### PR TITLE
Add optional preserveFontStyle

### PR DESCRIPTION
**Pull request by mrinalini-m**
In response to this feature requesst: https://github.com/octref/shiki/issues/15
Optional preserveFontStyle passed as a prop in 

```(js)
shiki
  .getHighlighter({
     theme: 'nord',
     preserveFontStyle: false
   })
```
I found the style maps from vscode-textmate but I didn't see a mapping function like colorMap. 
```(js)
 this._styleMap = {
    1: `font-style: italic;`, 
    2: `font-weight: bold`,
    4: `text-decoration: underline`
}
```

### CODE REVIEW

**Comment by h-2** [comment]
I would very much like to see this merged, thank you for creating this PR! Is there anything I can do to help get this in? I have just patched a local install, but my front-end (shiki-latex) doesn't automatically pick up the support...

File: package.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
     "packages/*"
   ],
   "name": "shiki",
+  "version": "0.1.7",
```
Comment (octref) [inline package.json:7]: Let's not do that because it's a yarn workspace.

File: README.md [hunk 2]
```diff
@@ -17,7 +17,8 @@ const shiki = require('shiki')
 
 shiki
   .getHighlighter({
-    theme: 'nord'
+    theme: 'nord',
+    preserveFontStyle: false //Optional. Defaults to true
```
Comment (octref) [inline README.md:21]: I need to think a bit about adding this in the high level API. For now I'd just leave that in the `codeToHtml` options.

**Review by octref — CHANGES_REQUESTED** [review]
Good work and sorry for leaving this stale for too long. I'll do the conflict resolution myself and fix some issues.

**Comment by octref** [comment]
I merged via https://github.com/shikijs/shiki/commit/3b9583519a456efab3b040039594e0f459bad340
Thanks for your contribution.

**Pull request closed without merging**